### PR TITLE
HPCC-30553 Add persistent tabs to "details" pages

### PR DESCRIPTION
**Pull request by GordonSmith**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/controls/TabPane.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    return (
+        <MenuItem key={tab.id} icon={tab.icon} onClick={onClick}>
+            <div>{tab.name}</div>
+        </MenuItem>
+    );
+};
+
+//----- OverflowMenu -----//
```
Comment (GordonSmith) [inline esp/src/src-react/components/controls/TabPane.tsx:88]: Remove comment

File: esp/src/src-react/components/controls/TabPane.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+const tabs: ExampleTab[] = [];
+
+type OverflowMenuItemProps = {
+    tab: ExampleTab;
+    onClick: React.MouseEventHandler<ARIAButtonElement<"div">>;
+};
+
+/**
+ * A menu item for an overflow menu that only displays when the tab is not visible
+ */
```
Comment (GordonSmith) [inline esp/src/src-react/components/controls/TabPane.tsx:72]: Remove comment

File: esp/src/src-react/components/controls/TabPane.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    },
+});
+
+type OverflowMenuProps = {
+    onTabSelect?: (tabId: string) => void;
+};
+
+/**
+ * A menu for selecting tabs that have overflowed and are not visible.
+ */
```
Comment (GordonSmith) [inline esp/src/src-react/components/controls/TabPane.tsx:105]: Remove comment

File: esp/src/src-react/components/controls/TabbedPanes/Count.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+interface CountProps {
+    value: string | number;
+}
+
+export const Count: React.FunctionComponent<CountProps> = ({
+    value
+}) => {
+
+    if (!value) return <></>;
+    return <span> ({value})</span>;
```
Comment (GordonSmith) [inline esp/src/src-react/components/controls/TabbedPanes/Count.tsx:12]: TODO:  Could be prettier

File: esp/src/src-react/components/controls/TabbedPanes/OverflowMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    if (isVisible) {
+        return null;
+    }
+
+    return (
+        <MenuItem key={tab.id} icon={tab.icon} disabled={tab.disabled} onClick={onClick}>
+            <div>{tab.label}<Count value={tab.count} /></div>
+        </MenuItem>
+    );
```
Comment (GordonSmith) [inline esp/src/src-react/components/controls/TabbedPanes/OverflowMenu.tsx:27]: Remove parens

File: esp/src/src-react/components/controls/TabbedPanes/OverflowMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    tab: TabInfo;
+    onClick: React.MouseEventHandler<ARIAButtonElement<"div">>;
+};
+
+const OverflowMenuItem = (props: OverflowMenuItemProps) => {
+    const { tab, onClick } = props;
+    const isVisible = useIsOverflowItemVisible(tab.id);
+
+    if (isVisible) {
+        return null;
```
Comment (GordonSmith) [inline esp/src/src-react/components/controls/TabbedPanes/OverflowMenu.tsx:20]: Return `<></>` for consistency

File: esp/src/src-react/components/controls/TabbedPanes/Count.tsx (hunk 4 above)
Comment (GordonSmith) [reply esp/src/src-react/components/controls/TabbedPanes/Count.tsx:12]: Future work

File: esp/src/src-react/components/controls/TabPane.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
+import * as React from "react";
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply esp/src/src-react/components/controls/TabPane.tsx]: Removed

File: esp/src/src-react/components/WorkunitDetails.tsx [hunk 8]
```diff
@@ -70,55 +55,112 @@ export const WorkunitDetails: React.FunctionComponent<WorkunitDetailsProps> = ({
         });
     }, [wuid], [queryParams]);
 
+    const onTabSelect = (id: string) => {
+        // setSelectedValue(data.value);
```
Comment (jeclrsg) [inline esp/src/src-react/components/WorkunitDetails.tsx:59]: some commented out code

**Review by jeclrsg — COMMENTED** [review]
@GordonSmith Just saw a couple spots w/ some old commented out code. Also, when I checked this branch out, I noticed that the state wasn't being passed in correctly when you changed tabs. Though, I couldn't remember if that was one of the things we'd talked about earlier today & you'd said you still had changes to make on that.

**Comment by GordonSmith** [comment]
@jeclrsg changes made as requested and added "tab state" support.

**Review by jeclrsg — APPROVED** [review]

**Pull request merged**